File: code/CS/Apps/MetricView/MetricView/ChartDirector/chartdir_net/ChartDirector/NetWPFCharts/NetWPFCharts/financedemo.cs
using System;
using ChartDirector;

namespace CSharpWPFDemo
{
  class financedemo : DemoModule
  {
    //Name of demo module
    public string getName() {
      return "Interactive Financial Chart";
    }

    //Number of charts produced in this demo module
    public int getNoOfCharts()
    {
      return 1;
    }

    //Main code for creating chart.
    public void createChart(WPFChartViewer viewer, int chartIndex)
    {
      //This demo uses its own window. The viewer on the right pane is not used.
      viewer.Chart = null;
      new FinanceDemoWindow().ShowDialog();
    }
  }
}

File: code/CS/Apps/MetricView/MetricView/ChartDirector/chartdir_net/ChartDirector/NetWinCharts/VBNetWinCharts/financedemo.vb
Imports ChartDirector

Public Class financedemo
    Implements DemoModule

    ' Name of demo module
    Public Function getName() As String Implements DemoModule.getName
        Return "Interactive Financial Chart"
    End Function

    ' Number of charts produced in this demo module
    Public Function getNoOfCharts() As Integer Implements DemoModule.getNoOfCharts
        Return 1
    End Function

    ' Main code for creating chart.
    Public Sub createChart(ByVal viewer As WinChartViewer, ByVal chartIndex As Integer) _
        Implements DemoModule.createChart
        'This demo uses its own form. The viewer on the right pane is not used.
        viewer.Image = Nothing
        Dim f As System.Windows.Forms.Form = New FrmFinanceDemo()
        f.ShowDialog()
        f.Dispose()
    End Sub

End Class
